### PR TITLE
Decouple tests from the workflow service container

### LINKED ISSUE #1675: Decouple testing from the workflow service
If we can stub out the workflow service, our test suite should be much faster as we don't have to download and spin up that container. Additionally there will be no network delay when doing indexing.  Currently if we turn off the workflow-service we have these errors:

```
Finished in 10 minutes 52 seconds (files took 5.87 seconds to load)
768 examples, 25 failures, 3 pending

Failed examples:

rspec ./spec/forms/apo_form_spec.rb:298 # ApoForm no model (new) #save hits the registration service to register both an APO and a collection
rspec ./spec/forms/collection_form_spec.rb:26 # CollectionForm creates a collection from title/abstract by registering the collection, then adding the abstract
rspec ./spec/controllers/items_controller_spec.rb:567 # ItemsController#create_obj_and_apo loads an APO object so that it has the appropriate model type (according to the solr doc)
rspec ./spec/controllers/items_controller_spec.rb:574 # ItemsController#create_obj_and_apo loads an Item object so that it has the appropriate model type (according to the solr doc)
rspec ./spec/integration/item_displays_spec.rb:66 # mods_view item dialogs add workflow renders the add workflow ui
rspec ./spec/features/release_history_spec.rb:10 # Release history items show a release history
rspec ./spec/features/release_history_spec.rb:17 # Release history adminPolicy objects do not show release history
rspec ./spec/controllers/versions_controller_spec.rb:96 # VersionsController#create_obj loads an APO object so that it has the appropriate model type (according to the solr doc)
rspec ./spec/controllers/versions_controller_spec.rb:106 # VersionsController#create_obj loads an Item object so that it has the appropriate model type (according to the solr doc)
rspec ./spec/features/bulk_jobs_spec.rb:11 # Bulk jobs view the submit button exists
rspec ./spec/features/consistent_titles_spec.rb:16 # Consistent titles catalog show view
rspec ./spec/controllers/collections_controller_spec.rb:82 # CollectionsController#create adds the collection to the apo default collection list
rspec ./spec/controllers/collections_controller_spec.rb:54 # CollectionsController#create creates a collection from title/abstract by registering the collection, then adding the abstract
rspec ./spec/controllers/collections_controller_spec.rb:33 # CollectionsController#create creates a collection via catkey
rspec ./spec/integration/apo_spec.rb:28 # apo creates and edits an apo
rspec ./spec/features/item_view_metadata_spec.rb:68 # Item view when the file is on the workspace can be downloaded
rspec ./spec/features/item_view_metadata_spec.rb:15 # Item view when the file is not on the workspace shows the file info
rspec ./spec/features/bulk_screen_spec.rb:14 # Bulk actions view basic page renders ok
rspec ./spec/views/workflows/_new.html.erb_spec.rb:6 # workflows/_new.html.erb renders the partial content
rspec ./spec/features/report_view_spec.rb:19 # Report view bulk returns a page with the expected elements
rspec ./spec/features/item_source_id_change_spec.rb:26 # Item source id change when modification is allowed changes the source id
rspec ./spec/controllers/content_types_controller_spec.rb:30 # ContentTypesController#update with access is successful
rspec ./spec/controllers/content_types_controller_spec.rb:46 # ContentTypesController#update with access in a batch process is successful
rspec ./spec/controllers/content_types_controller_spec.rb:39 # ContentTypesController#update with access and an invalid content_type is forbidden
rspec ./spec/features/item_catkey_change_spec.rb:26 # Item catkey change when modification is allowed changes the catkey
```

Success criteria: 
Remove workflow-service from `.travis.yml` and the tests pass.

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #1675

## Why was this change made?
This should improve the speed of the tests because we don't need to download or run an extra container.

## Was the documentation updated?

n/a

### CODE REVIEW

File: spec/controllers/items_controller_spec.rb [hunk 1]
```diff
@@ -563,23 +561,6 @@
...
-      expect(Dor).to receive(:find).with('druid:zt570tx3016').and_call_original # override the earlier Dor.find expectation
-      allow(Dor).to receive(:find).with('druid:hv992ry2431') # create_obj_and_apo will try to lookup the APO's APO
-      subject.send(:create_obj_and_apo, 'druid:zt570tx3016')
-      expect(subject.instance_variable_get(:@object).to_solr).to include('active_fedora_model_ssi' => 'Dor::AdminPolicyObject',
-                                                                         'has_model_ssim' => 'info:fedora/afmodel:Dor_AdminPolicyObject')
-    end
-    it 'loads an Item object so that it has the appropriate model type (according to the solr doc)' do
-      expect(Dor).to receive(:find).with('druid:hj185vb7593').and_call_original # override the earlier Dor.find expectation
-      allow(Dor).to receive(:find).with('druid:ww057vk7675') # create_obj_and_apo will try to lookup the Item's APO
-      subject.send(:create_obj_and_apo, 'druid:hj185vb7593')
```
Comment (jcoyne) [inline spec/controllers/items_controller_spec.rb]: This was testing a private method that no longer exists

File: spec/controllers/items_controller_spec.rb [hunk 2]
```diff
@@ -563,23 +561,6 @@
     end
   end
 
-  describe '#create_obj_and_apo' do
-    it 'loads an APO object so that it has the appropriate model type (according to the solr doc)' do
-      expect(Dor).to receive(:find).with('druid:zt570tx3016').and_call_original # override the earlier Dor.find expectation
-      allow(Dor).to receive(:find).with('druid:hv992ry2431') # create_obj_and_apo will try to lookup the APO's APO
-      subject.send(:create_obj_and_apo, 'druid:zt570tx3016')
```
Comment (jcoyne) [inline spec/controllers/items_controller_spec.rb]: This was testing a private method that no longer exists

File: spec/controllers/versions_controller_spec.rb [hunk 3]
```diff
@@ -91,26 +91,4 @@
       end
     end
   end
-
-  describe '#create_obj' do
```
Comment (jcoyne) [inline spec/controllers/versions_controller_spec.rb]: This was testing a private method that is covered by other tests.

File: app/forms/apo_form.rb [hunk 4]
```diff
@@ -35,6 +35,7 @@ def sync
     model.mods_title           = params[:title]
     model.desc_metadata_format = params[:desc_md]
     model.metadata_source      = params[:metadata_source]
+
```
Comment (mjgiarlo) [inline app/forms/apo_form.rb]: :question:

Comment (jcoyne) [reply app/forms/apo_form.rb]: Oops.  I probably threw a byebug here at one time and then remove it but left the `\n`

Comment (mjgiarlo) [reply app/forms/apo_form.rb]: nbd

**Pull request merged**